### PR TITLE
fix testnet zilbridge errors

### PR DESCRIPTION
**Pull request by JaydenTeoh**

### CODE REVIEW

File: src/util/blockchain.ts [hunk 1]
```diff
@@ -126,63 +126,139 @@ export const getBlockchainFromChain = (chainId?: number) => {
...
+  } else {
+    switch (network) {
+      case Network.MainNet:
+        switch (chainId) {
+          case 0:
+            return Blockchain.Native
+          case 1:
+            return Blockchain.Btc
+          case 4:
+            return Blockchain.Carbon
```
Comment (sarah-thong) [inline src/util/blockchain.ts:198]: On Carbon chain, chain-id 4 is Neo Legacy (refer to [Bridges API](https://api.carbon.network/carbon/coin/v1/bridges?pagination.limit=10000)). Where did you get chain-id 4 for Carbon?
Screenshot 2023-01-31 at 9 10 53 PM

File: src/util/blockchain.ts [hunk 2]
```diff
@@ -126,63 +126,139 @@ export const getBlockchainFromChain = (chainId?: number) => {
...
+        }
+      case Network.DevNet:
+        switch (chainId) {
+          case 350:
+            return Blockchain.Ethereum
+          case 5:
+            return Blockchain.Neo
+          case 79:
+            return Blockchain.BinanceSmartChain
+        }
```
Comment (sarah-thong) [inline src/util/blockchain.ts:257]: Remember to include all blockchains in the [devnet Bridges API](https://dev-api.carbon.network/carbon/coin/v1/bridges?pagination.limit=10000)

File: src/util/blockchain.ts [hunk 3]
```diff
@@ -126,63 +126,139 @@ export const getBlockchainFromChain = (chainId?: number) => {
...
+      case Network.TestNet:
+        switch (chainId) {
+          case 5:
+            return Blockchain.Carbon
+          case 88:
+            return Blockchain.Neo3
+          case 111:
+            return Blockchain.Zilliqa
+          case 502:
+            return Blockchain.Ethereum
```
Comment (sarah-thong) [inline src/util/blockchain.ts:245]: Don't forget to include chain_id 2 (Ethereum)
https://test-api.carbon.network/carbon/coin/v1/bridges?pagination.limit=10000

File: src/util/blockchain.ts [hunk 4]
```diff
@@ -126,63 +126,139 @@ export const getBlockchainFromChain = (chainId?: number) => {
...
+          case 317:
+            return Blockchain.Comdex
+          case 318:
+            return Blockchain.StafiHub
+          case 319:
+            return Blockchain.Persistence
+          case 320:
+            return Blockchain.Stargaze
+          case 42161:
+            return Blockchain.Arbitrum
```
Comment (sarah-thong) [inline src/util/blockchain.ts:232]: 2 instances of Arbitrum in this switch statement, can simplify into
```
switch (...) {
  ...
  case 19:
  case 42161:
    return Blockchain.Arbitrum
}
```

File: src/util/blockchain.ts [hunk 5]
```diff
@@ -126,63 +126,139 @@ export const getBlockchainFromChain = (chainId?: number) => {
...
+          default:
+            return undefined
+        }
```
Comment (sarah-thong) [inline src/util/blockchain.ts:235]: Got a couple of blockchains listed on [Bridges API](https://api.carbon.network/carbon/coin/v1/bridges?pagination.limit=10000) that are missing in here (e.g. Neo3/Zilliqa/Ethereum). Do include under the Mainnet clause.

File: src/util/blockchain.ts [hunk 6]
```diff
@@ -73,116 +73,149 @@ export const CHAIN_IDS: ChainIds = {
...
-    case 42161: // mainnnet
-      return Blockchain.Arbitrum;
+export const blockchainForChainId = (chainId?: number, network = Network.MainNet): Blockchain | undefined => {
+  switch (network) {
+    case Network.MainNet:
+      switch (chainId) {
+        case 0:
+          return Blockchain.Native
+        case 1:
+          return Blockchain.Btc
```
Comment (sarah-thong) [inline src/util/blockchain.ts:136]: Don't forget to add Eth under `case Network.Mainnet`
```
i.e.
case 2:
  return Blockchain.Ethereum
```

File: src/util/blockchain.ts [hunk 7]
```diff
@@ -73,116 +73,149 @@ export const CHAIN_IDS: ChainIds = {
...
+        case 4:
+          return Blockchain.Neo
+        case 6:
+          return Blockchain.BinanceSmartChain
+        case 14:
+          return Blockchain.Neo3
+        case 9:  /* FALLTHROUGH */
+        case 18: /* FALLTHROUGH */
+        case 111:
```
Comment (sarah-thong) [inline src/util/blockchain.ts:145]: If I'm not wrong, `111` is for testnet

File: src/util/blockchain.ts [hunk 8]
```diff
@@ -73,116 +73,149 @@ export const CHAIN_IDS: ChainIds = {
...
+        case 19: /* FALLTHROUGH */
+        case 42161:
+          return Blockchain.Arbitrum
+        default:
+          return undefined
+      }
+    case Network.TestNet:
+      switch (chainId) {
+        case 1:
+          return Blockchain.Terra
```
Comment (sarah-thong) [inline src/util/blockchain.ts:191]: Hmm on Carbon chain, chain-id 1 is BTC too. I'm not quite sure if this is going to work out... Might wanna check with Steven how to deal with this
Screenshot 2023-02-01 at 3 38 30 PM

File: src/util/blockchain.ts [hunk 9]
```diff
@@ -73,116 +73,149 @@ export const CHAIN_IDS: ChainIds = {
...
+        case 5:
+          return Blockchain.Carbon
+        case 79:
+          return Blockchain.BinanceSmartChain
+        case 88:
+          return Blockchain.Neo3
+        case 111:
+          return Blockchain.Zilliqa
+        case 502:
+          return Blockchain.Ethereum
```
Comment (sarah-thong) [inline src/util/blockchain.ts:203]: For Ethereum, the statement can be shortened as such:
```
case 2:
case 502:
  return Blockchain.Ethereum
```

File: src/util/blockchain.ts (hunk 8 above)
Comment (stevenkhong) [reply src/util/blockchain.ts:204]: why is 1 = terra on testnet? looks like a mistake?

**Review by stevenkhong — CHANGES_REQUESTED** [review]

File: src/util/blockchain.ts (hunk 8 above)
Comment (JaydenTeoh) [reply src/util/blockchain.ts:204]: @stevenkhong chainId 1 via polynetwork is BTC, via IBC is Terra, not sure which to choose. [testnet bridges api](https://test-api.carbon.network/carbon/coin/v1/bridges?pagination.limit=10000)

**Pull request merged**